target/linux/x86/base-files/usr/sbin/grow_rootfspart

```diff
@@ -0,0 +1,327 @@
+#!/bin/sh
+# shellcheck disable=SC2039
+#
+# Grow partition 2 (rootfs_data) until the end of the disk
+#
+# 
+
+readbin2hex() {
+	local input=$1
+	local seek=$2
+	local length=$3
+	hexdump -ve '1/1 "%.2x"' -s "$seek" -n "$length" "$input"
+}
+
+hex2bin(){
+	printf "%s\n" "$1" | sed -r 's/(.{2})/\\\\x\1 /g' | xargs printf '%b' 
+}
+
+writehex2bin(){
+	local output=$1
+	local seek=$2
+	local hexbuf=$3
+	local length=$((${#hexbuf}/2))
+	hex2bin "$hexbuf" | dd bs=1 seek="$seek" count="$length" of="$output" conv=notrunc 2>/dev/null
+}
+writehex2buf(){
+	local buf=$1
+	local seek=$2
+	local modified=$3
+	printf "%s" "${buf:0:seek*2}"
+	printf "%s" "$modified"
+	printf "%s\n" "${buf:seek*2+${#modified}}"
+}
+
+crc32_hex(){
+	gzip -c | tail -c8 | hexdump -n4 -e '"%.8x" "\n"'
```

Comment:
consider making this '/bin/busybox gzip ...' for robustness

Reply:
Both, busybox or standalone, will work as expected. If I add a busybox, it might fail when there is no busybox installed.
This script can be executed from the host modifying the img file, where busybox might not be available. I would keep it as it is.

Reply:
yeah... should be ok @ preinit... there is a bug at the moment @ stage2 + gzip full breaking zcat ( for devices using img.gz sysupgrade... agree that it likely wont effect this... ( it's really a stage2 issue / gzip full package issue ) but felt it would be good to mention it.

Reply:
Thanks for the mention. I guess sysupgrade is a much more complex env as it creates a ramfs from a running system.
If you have multiple implementations for a program, it might get the wrong file.

In preinit, even if gzip is builtin, it should not be a problem as both works as expected.